Memfault: Add support for Wi-Fi and nRF7002 DK, move sample to `debug/` (#11019)
nrfconnect/sdk-nrf: samples/debug/memfault/README.rst

Coredumps
=========

The sample enables Memfault shell by default.

> The shell offers multiple commands to test a wide range of functionality offered by Memfault SDK.

Reviewer: ```suggestion
The shell offers multiple commands to test a wide range of functionality offered by the Memfault SDK.
```

=============

To send data to the Memfault cloud, a project key must be configured using :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY`.

> If nRF7002 DK is used and one wishes to automatically connect to a Wi-Fi network upon boot, the Wi-Fi credentials be configured using :kconfig:option`CONFIG_WIFI_CREDENTIALS_STATIC_SSID` and :kconfig:option:`CONFIG_WIFI_CREDENTIALS_STATIC_PASSWORD`.

Reviewer: ```suggestion
If you are using the nRF7002 DK and want to automatically connect to a Wi-Fi network upon boot, configure the Wi-Fi credentials using the Kconfig options :kconfig:option`CONFIG_WIFI_CREDENTIALS_STATIC_SSID` and :kconfig:option:`CONFIG_WIFI_CREDENTIALS_STATIC_PASSWORD`.
```

        <inf> mflt: GNU Build ID: a09094cdf9da13f20719f87016663ab529b71267
        <inf> memfault_sample: Memfault sample has started

#. The sample connects to an available LTE or Wi-Fi network, which is indicated by the following message:

   .. code-block:: console

        <inf> memfault_sample: Connecting to network

> #. When the connection is established, the following messag eis printed:

Reviewer: ```suggestion
#. When the connection is established, the following message is shown:
```
